modin/pandas/test/test_io.py

```diff
@@ -995,12 +995,21 @@ def test_read_csv_skiprows_names(self, names, skiprows):
             skiprows=skiprows,
         )
 
+    def _has_pandas_fallback_reason(self):
+        # The Python engine does not use custom IO dispatchers, so specialized error messages
+        # won't appear
+        return Engine.get() != "Python" and StorageFormat.get() != "Omnisci"
```

Comment:
OmnisciOnNativeIO can read a CSV file with pyarrow and also can be defaulting to pandas. Doesn't it seem to you that the condition should be corrected?

Reply:
I believe adding Omnisci to this condition was necessary because `OmnisciOnNativeIO` manually defines [`read_csv`](https://github.com/modin-project/modin/blob/master/modin/experimental/core/execution/native/implementations/omnisci_on_native/io/io.py#L161) instead of using the implementation from `TextFileDispatcher` (which matters because of [this line](modin/pandas/test/dataframe/test_binary.py::test_math_functions_level[add]), which produces error messages). This is different form the Ray backend, for example, where [this line](modin/pandas/test/dataframe/test_binary.py::test_math_functions_level[add]) leads to the TextFileDispatcher method being used.

It does seem like Omnisci does its own [error validation](https://github.com/modin-project/modin/blob/master/modin/experimental/core/execution/native/implementations/omnisci_on_native/io/io.py#L403), so perhaps this test should be more refined to check that error message on the Omnisci format, instead of the standard TextFileDispatcher one.